ckan/lib/uploader.py

```diff
@@ -277,25 +277,25 @@ def upload(self, id, max_size=10):
                 if e.errno != 17:
                     raise
             tmp_filepath = filepath + '~'
-            output_file = open(tmp_filepath, 'wb+')
-            self.upload_file.seek(0)
-            current_size = 0
-            while True:
-                current_size = current_size + 1
-                # MB chunks
-                data = self.upload_file.read(2 ** 20)
-
-                if not data:
-                    break
-                output_file.write(data)
-                if current_size > max_size:
-                    os.remove(tmp_filepath)
-                    raise logic.ValidationError(
-                        {'upload': ['File upload too large']}
-                    )
+            with open(tmp_filepath, 'wb+') as output_file:
+                with self.upload_file:
+                    self.upload_file.seek(0)
```

Comment:
But what if `self.upload_file` is a StringIO object(this is case for test and, probably, that could be used for programmatical resource creation)? In that case you need to reset pointer and then copy buffer content into file